lib/resources/resourceBase.js

```diff
@@ -33,6 +33,9 @@ class ResourceBase {
     let isMultiPartForm = false;
 
     for (let key in form) {
+      if (form[key] === undefined) {
+        throw new Error(`${key} is undefined`);
```

Comment:
Let's change this to `delete form[key]` instead of throwing an error.
Since we don't throw an error on undefined values in the payload when it's not a multipart form, we'd prefer to have the same behavior for multipart forms.